Dockerfile.assisted-service

```diff
@@ -47,6 +48,7 @@ RUN curl -s $OC_URL | tar -xzC /usr/local/bin/ oc
 COPY --from=builder /build/assisted-service /assisted-service
 COPY --from=builder /build/assisted-service-operator /assisted-service-operator
 COPY --from=builder /build/assisted-service-admission /assisted-service-admission
+COPY --from=builder /build/ephemeral-installer-register-cluster-and-infraenv /ephemeral-installer-register-cluster-and-infraenv
```

Comment:
As soon as https://github.com/openshift/assisted-service/pull/3660 will land it will be necessary to add the ephemeral client in `Dockerfile.ocp` as well. 
Not sure if in future (when the images will be shipped in the ocp payload) will make sense to keep that binary in the `Dockerfile.assisted-service` as well